spot_driver/test/src/conversions/test_robot_state.cpp

```diff
@@ -0,0 +1,844 @@
+  ASSERT_THAT(out->velocity, SizeIs(2));
+  ASSERT_THAT(out->effort, SizeIs(2));
+
+  EXPECT_THAT(out->header, ClockSkewIsAppliedToHeader(timestamp, clock_skew));
+
+  // THEN the joint names were converted to the "friendly" names used in the URDF
+  ASSERT_THAT(out->name, UnorderedElementsAre("my_prefix/front_left_hip_x", "my_prefix/arm_wr0"));
+
+  // THEN the position, velocity, and effort corresponding to each named joint match the input joint states
+  const std::size_t index1 =
+      std::distance(out->name.cbegin(), std::find(out->name.cbegin(), out->name.cend(), "my_prefix/front_left_hip_x"));
+  EXPECT_THAT(out->position.at(index1), DoubleEq(0.1));
+  EXPECT_THAT(out->velocity.at(index1), DoubleEq(0.2));
+  EXPECT_THAT(out->effort.at(index1), DoubleEq(0.4));
+
+  const std::size_t index2 =
+      std::distance(out->name.cbegin(), std::find(out->name.cbegin(), out->name.cend(), "my_prefix/arm_wr0"));
+  EXPECT_THAT(out->position.at(index2), DoubleEq(0.5));
+  EXPECT_THAT(out->velocity.at(index2), DoubleEq(0.6));
+  EXPECT_THAT(out->effort.at(index2), DoubleEq(0.8));
+}
+
+TEST(RobotStateConversions, TestGetJointStatesNoJointStates) {
+  // GIVEN a RobotState that does not contain any joint states, but does contains some other kinematic state info
+  ::bosdyn::api::RobotState robot_state;
+  google::protobuf::Timestamp timestamp;
+  timestamp.set_seconds(15);
+  timestamp.set_nanos(0);
+  robot_state.mutable_kinematic_state()->mutable_acquisition_timestamp()->CopyFrom(timestamp);
+
+  // GIVEN some nominal clock skew and prefix
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+  constexpr auto prefix = "prefix/";
+
+  // WHEN we create a JointState ROS message from the RobotState
+  const auto out = getJointStates(robot_state, clock_skew, prefix);
+
+  // THEN the message is created and contains a timestamp, but does not contain any joint state data
+  ASSERT_THAT(out.has_value(), IsTrue());
+  EXPECT_THAT(out->header, ClockSkewIsAppliedToHeader(timestamp, clock_skew));
+  EXPECT_THAT(out->name, IsEmpty());
+  EXPECT_THAT(out->position, IsEmpty());
+  EXPECT_THAT(out->velocity, IsEmpty());
+  EXPECT_THAT(out->effort, IsEmpty());
+}
+
+TEST(RobotStateConversions, TestGetJointStatesNoKinematicState) {
+  // GIVEN a RobotState that does not contain any kinematic state data whatsoever
+  ::bosdyn::api::RobotState robot_state;
+
+  // GIVEN some nominal clock skew and prefix
+  google::protobuf::Duration clock_skew;
+  clock_skew.set_seconds(1);
+  constexpr auto prefix = "prefix/";
+
+  // WHEN we attempt to create a JointState ROS message from the RobotState
+  const auto out = getJointStates(robot_state, clock_skew, prefix);
+
+  // THEN no ROS message is output
+  ASSERT_THAT(out.has_value(), IsFalse());
+}
+
+TEST(RobotStateConversions, TestGetTf) {
+  // GIVEN a RobotState containing a valid transform snapshot
+  ::bosdyn::api::RobotState robot_state;
+  // GIVEN a nonzero acquisition timestamp
```

Comment:
Optionally, you can write
```suggestion
  // AND a nonzero acquisition timestamp
```
which is still valid gherkin